pkg/cloud/vsphere/services/govmomi/extra/config.go

```diff
@@ -38,13 +38,11 @@ func (e *Config) SetCloudInitUserData(data []byte) error {
 			Value: "base64",
 		},
 	)
-
-	return nil
 }
 
 // SetCloudInitMetadata sets the cloud init user data at the key
 // "guestinfo.metadata" as a base64-encoded string.
-func (e *Config) SetCloudInitMetadata(data []byte) error {
+func (e *Config) SetCloudInitMetadata(data []byte) {
```

Comment:
Hi @vijaykatam,

Please remove this and the other function signature change as they are not integral to this patch. Feel free to open a separate PR for the functions. Thank you!